applications/DamApplication/custom_utilities/construction_utility.hpp

```diff
@@ -88,99 +88,75 @@ class ConstructionUtility
         mThermalLastCondition = mrThermalModelPart.GetMesh(0).Conditions().size();
 
+        const ProcessInfo& r_current_process_info = mrMechanicalModelPart.GetProcessInfo();
+
         if (nelements != 0)
         {
             ModelPart::ElementsContainerType::iterator el_begin = mrMechanicalModelPart.ElementsBegin();
-            ModelPart::ElementsContainerType::iterator el_begin_thermal = mrThermalModelPart.ElementsBegin();
             mNumNode = el_begin->GetGeometry().PointsNumber();
 
-            #pragma omp parallel for
-            for (int k = 0; k < nelements; ++k)
-            {
-                ModelPart::ElementsContainerType::iterator it = el_begin + k;
-                ModelPart::ElementsContainerType::iterator it_thermal = el_begin_thermal + k;
-                it->Set(ACTIVE, false);
-                it_thermal->Set(ACTIVE, false);
-            }
+            block_for_each(mrMechanicalModelPart.Elements(), [&](auto& rMechanicalElement){
+                rMechanicalElement.Set(ACTIVE, false);
+                rMechanicalElement.Initialize(r_current_process_info);
+            });
+
+            block_for_each(mrThermalModelPart.Elements(), [&](auto& rThermalElement){
+                rThermalElement.Set(ACTIVE, false);
+            });
 
             // Same nodes for both computing model part
-            ModelPart::NodesContainerType::iterator it_begin = mrThermalModelPart.NodesBegin();
-            #pragma omp parallel for
-            for (int i = 0; i < nnodes; ++i)
-            {
-                ModelPart::NodesContainerType::iterator it = it_begin + i;
-                it->Set(ACTIVE, false);
-                it->Set(SOLID, false);
-            }
+            block_for_each(mrThermalModelPart.Nodes(), [&](auto& rNode){
+                rNode.Set(ACTIVE, false);
+                rNode.Set(SOLID, false);
+            });
         }
 
         // Activation of the existing parts, either the soil or the already built dam ( User must specify each part through the interface)
         if (mActivateSoilPart == true)
         {
             const int soil_nelements = mrMechanicalModelPart.GetSubModelPart(mMechanicalSoilPart).Elements().size();
-            const int soil_nnodes = mrMechanicalModelPart.GetSubModelPart(mMechanicalSoilPart).Nodes().size();
 
             if (soil_nelements != 0)
             {
-                ModelPart::ElementsContainerType::iterator el_begin = mrMechanicalModelPart.GetSubModelPart(mMechanicalSoilPart).ElementsBegin();
-                ModelPart::ElementsContainerType::iterator el_begin_thermal = mrThermalModelPart.GetSubModelPart(mThermalSoilPart).ElementsBegin();
-                mNumNode = el_begin->GetGeometry().PointsNumber();
+                block_for_each(mrMechanicalModelPart.GetSubModelPart(mMechanicalSoilPart).Elements(), [&](auto& rMechanicalElement){
+                    rMechanicalElement.Set(ACTIVE, true);
+                });
```

Comment:
I think this function already exists in the VariableUtils

Reply:
Great advice, thank you! :)